commons-vfs2-sandbox/src/main/java/org/apache/commons/vfs2/provider/smb2/Smb2ClientWrapper.java

```diff
@@ -0,0 +1,223 @@
+        } catch (Exception e) {
+            throw new FileSystemException("vfs.provider.smb2/connect.error", rootName.getHostName(), e.getCause());
+        }
+    }
+
+    public GenericFileName getRoot() {
+
+        return root;
+    }
+
+    /**
+     * Returns file information of a given file.
+     *
+     * @param relativePath of file
+     * @return FileAllInformation object
+     */
+    public FileAllInformation getFileInfo(String relativePath) {
+
+        try {
+            return diskShare.getFileInformation(relativePath);
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    /**
+     * Returns a write handle on the file.
+     *
+     * @param path path to file
+     * @return DiskEntry file
+     */
+    public DiskEntry getDiskEntryWrite(String path) {
+
+        return diskShare.open(path,
+                EnumSet.of(AccessMask.MAXIMUM_ALLOWED),
+                EnumSet.of(FileAttributes.FILE_ATTRIBUTE_NORMAL),
+                EnumSet.of(SMB2ShareAccess.FILE_SHARE_WRITE),
+                SMB2CreateDisposition.FILE_OPEN_IF,
+                EnumSet.of(SMB2CreateOptions.FILE_NO_COMPRESSION));
+    }
+
+    /**
+     * Creates a folder on a given path.
+     *
+     * @param path of folder
+     */
+    public void createFolder(String path) {
+
+        DiskEntry de = getDiskEntryFolderWrite(path);
+        de.close();
+    }
+
+    /**
+     * Returns a handle for a folder.
+     *
+     * @param path folder path
+     * @return DiskEntry
+     */
+    public DiskEntry getDiskEntryFolderWrite(String path) {
+
+        return diskShare.openDirectory(path,
+                EnumSet.of(AccessMask.GENERIC_ALL),
+                EnumSet.of(FileAttributes.FILE_ATTRIBUTE_NORMAL),
+                EnumSet.of(SMB2ShareAccess.FILE_SHARE_READ),
+                SMB2CreateDisposition.FILE_OPEN_IF,
+                EnumSet.of(SMB2CreateOptions.FILE_DIRECTORY_FILE));
+    }
+
+    /**
+     * Returns a read handle on the file.
+     *
+     * @param path path to file
+     * @return DiskEntry file
+     */
+    public DiskEntry getDiskEntryRead(String path) {
+
+        return diskShare.open(path,
+                EnumSet.of(AccessMask.GENERIC_READ),
+                EnumSet.of(FileAttributes.FILE_ATTRIBUTE_NORMAL),
+                EnumSet.of(SMB2ShareAccess.FILE_SHARE_READ),
+                SMB2CreateDisposition.FILE_OPEN,
+                EnumSet.of(SMB2CreateOptions.FILE_NO_COMPRESSION));
+    }
+
+    /**
+     * Returns directory listing of path.
+     *
+     * @param path
+     * @return listing string array
+     */
+    public String[] getChildren(String path) {
+
+        List<String> children = new ArrayList<>();
+
+        for (FileIdBothDirectoryInformation file : diskShare.list(path)) {
+            String name = file.getFileName();
+            if (name.equals(".") || name.equals("..") || name.equals("./") || name.equals("../")) {
+                continue;
```

Comment:
Shall we add a comment why we are skipping

Reply:
done